samples/apps/copilot-chat-app/webapi/Program.cs

```diff
@@ -139,26 +160,42 @@ private static void AddSemanticKernelServices(this IServiceCollection services,
             return NullMemory.Instance;
         });
 
-        // Add persistent storage
-        // InMemory version
-        var chatSessionInMemoryContext = new InMemoryContext<ChatSession>();
-        var chatMessageInMemoryContext = new InMemoryContext<ChatMessage>();
+        // Add persistent chat storage
+        IStorageContext<ChatSession> chatSessionInMemoryContext;
+        IStorageContext<ChatMessage> chatMessageInMemoryContext;
+
+        switch (configuration["ChatStore:Type"].ToUpperInvariant())
+        {
+            case "VOLATILE":
+                chatSessionInMemoryContext = new VolatileContext<ChatSession>();
+                chatMessageInMemoryContext = new VolatileContext<ChatMessage>();
+                break;
+
+            case "FILESYSTEM":
+                FileSystemConfig filesystemConfig = configuration.GetSection("ChatStore:Filesystem").Get<FileSystemConfig>();
+                string fullPath = Path.GetFullPath(filesystemConfig.FilePath);
+                string directory = Path.GetDirectoryName(fullPath) ?? string.Empty;
+                chatSessionInMemoryContext = new FileSystemContext<ChatSession>(
+                    new FileInfo(Path.Combine(directory, $"{Path.GetFileNameWithoutExtension(fullPath)}_sessions{Path.GetExtension(fullPath)}")));
+                chatMessageInMemoryContext = new FileSystemContext<ChatMessage>(
+                    new FileInfo(Path.Combine(directory, $"{Path.GetFileNameWithoutExtension(fullPath)}_messages{Path.GetExtension(fullPath)}")));
+                break;
+
+            case "COSMOS":
+                CosmosConfig cosmosConfig = configuration.GetSection("ChatStore:Cosmos").Get<CosmosConfig>();
+#pragma warning disable CA2000 // Dispose objects before losing scope - objects are singletons for the duration of the process and disposed when the process exits.
+                chatSessionInMemoryContext = new CosmosDbContext<ChatSession>(
+                    cosmosConfig.ConnectionString, cosmosConfig.Database, "chat_sessions");
```

Comment:
Do we want to hard code the container name here? Could we move it to the config as well?

Reply:
At a minimum we would need to switch the hard coded name to one that does not have an underscore, see https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/how-to-dotnet-create-container for allowed chars in name.

Reply:
Moved to config and removed underscore.